### DISCUSSION TITLE
Clignotement d'un texte

### DISCUSSION THREAD

**Discussion by Bgtb** [witsa/synapps#763] [Q&A]
Bonjour,
Je cherche à faire clignoter un texte en fonction en fonction d'une variable ressource.

Pouvez-vous m'indiquer la procédure ?

Merci

**Comment by JeremieDENAEGEL** [comment]
Bonjour,
Ce composite devrait pouvoir répondre à ton soucis ;)
Il te faudra en ajouter un pour chaque élément que tu souhaites faire clignoter ;)

`SYNAPPS-STUDIO-COMPOSITE|{"config":{"key":"Defaut","name":"Point en defaut / Défaut en cours","properties":{"borderWidth":"0px","height":"0px","width":"0px"},"additionalDefs":{"isDefault":{"type":"boolean","label":"En défaut?","value":false},"targetKey":{"type":"actor","label":"Acteur ciblé","value":""},"rc":{"type":"color","label":"Couleur normale","value":null},"c":{"type":"color","label":"Couleur en défaut","value":"rgba(198, 77, 77, 0.68)"},"duration":{"type":"number","label":"Durée clignotement","value":600,"min":0,"step":100,"unit":"ms"}}},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"display/html","key":"default-style","properties":{"content":"<style>\n  @keyframes {{uniqKey}}_is_in_default {\n    from {\n      background-color: none;\n    }\n\n    to {\n      background-color: {{c}};\n    }\n  }\n</style>"},"additionalDefs":{"c":{"type":"color","label":"Couleur en défaut"},"isDefault":{"type":"boolean","label":"En défaut?"},"rc":{"type":"color","label":"Couleur normale"},"targetKey":{"type":"actor","label":"Acteur ciblé"},"duration":{"type":"number","label":"Durée clignotement","min":0,"step":100,"unit":"ms"},"uniqKey":{"type":"text"}},"additionals":{"c":"rgba(198, 77, 77, 0.68)","rc":"rgba(37, 207, 48, 0.42)","duration":600},"bindings":{"additionals.c":"stage@properties.c","additionals.targetKey":"stage@properties.targetKey","additionals.isDefault":"stage@properties.isDefault","additionals.rc":"stage@properties.rc","additionals.duration":"stage@properties.duration"},"events":{"onPropertyChanged":["","if ([","  'additionals.targetKey',","  'additionals.duration',","  'additionals.isDefault',","  'additionals.rc',","].includes(context.propertyPath)) {","","  if (context.propertyPath === 'additionals.targetKey') {","    const oldTargetKey = context.oldValue;","    if (!context.utils.isEmpty(oldTargetKey)) {","      const oldTarget = context.stage.stage.getActor(oldTargetKey);","      if (oldTarget) {","        oldTarget.properties.reset('animation');","        oldTarget.properties.reset('backgroundColor');","      }","    }","  }","","  this.runEvent('onPostInit');","}"],"onPostInit":["const targetKey = this.additionals.targetKey;","if (context.utils.isEmpty(targetKey)) return;","const target = context.stage.stage.getActor(targetKey);","if (context.utils.isNone(target)) return;","this.additionals.uniqKey = `${target.uniqClass}_${this.stage.uniqClass}`;","if (this.additionals.isDefault) {","  target.properties.animation = `${this.additionals.duration}ms linear infinite alternate ${this.additionals.uniqKey}_is_in_default`;","  target.properties.reset('backgroundColor');","} else {","  target.properties.reset('animation');","  target.properties.backgroundColor = this.additionals.rc;","}"]}}]}}`
